core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionNetworkOutput.java

```diff
@@ -176,6 +176,9 @@ public boolean send(Transmission transmission) {
                 if (ioe instanceof ConnectTimeoutException) {
                     transmissionPolicyManager.suspendInSeconds(TransmissionPolicy.BLOCKED_BUT_CAN_BE_PERSISTED, DEFAULT_BACKOFF_TIME_SECONDS);
                 }
+            } catch (IllegalStateException e) {
+                suspendTransmissions(TransmissionPolicy.BLOCKED_AND_CANNOT_BE_PERSISTED, response);
```

Comment:
It looks like `suspendTransmission()` will exit at the very first line because in case of the `IllegalStateException` - there is no `RetryAfter` header. This would mean that the behavior will be just to return here which is not different from general `catch (Exception e)` block. Are you sure that this suspension fixes the issue with the forever loop?